### ISSUE TITLE
Template field in templates field name vagueness

### ISSUE THREAD

**Issue by davebarda** [elastic/elasticsearch#18922]
**Description of the problem including expected versus actual behavior**:
The short explaination:
The field template In templates is vague because it is pattern matching the indices which will be working on, I think that the name should be somthing like Index pattern or so.

The Long one:
First to mention, I have started to use elasticsearch today,
I have used a twitter template which I couldn't understand why is not working.
I have searched the template file maybe It contains the index name and thats the reason It didn't work , I have noticed that when I changed the index name to the template name everything was just fine so I understood that I need to write about Index templates and I have understood that there is an index pattern matching with the template name.

**Comment by jpountz** [comment]
Do you have any recommendation?

**Comment by davebarda** [comment]
Well as I have wrote,
I think that index pattern should be suggestive, 
It will make people are new to templates to check what is all about.

**Comment by jpountz** [comment]
OK, so something like:

```
PUT /_template/template_1
{
  "index_pattern": "te*",
  "settings": {
    "number_of_shards": 1
  }
}
```

rather than

```
PUT /_template/template_1
{
  "template": "te*",
  "settings": {
    "number_of_shards": 1
  }
}
```

**Comment by clintongormley** [comment]
I agree - I've never liked that template parameter.

**Comment by clintongormley** [comment]
Depending on which gets in first, also see https://github.com/elastic/elasticsearch/pull/17300

**Comment by nik9000** [comment]
Removing `adoptme` because we have an open PR for this: #21009

**Comment by clintongormley** [comment]
Closed by #21009
